lib/web3/contract.js

```diff
@@ -57,7 +56,7 @@ var addFunctionsToContract = function (contract, abi) {
     abi.filter(function (json) {
         return json.type === 'function';
     }).map(function (json) {
-        return new SolidityFunction(json, contract.address);
+        return new SolidityFunction(contract._web3, json, contract.address);
```

Comment:
All the things we call solidly, are actually not solidly specific and work well for other languages like LLLL, as they are byte encoded EVM code. I suggest we rename all those to `evmEncoding` or something.